src/ui/components/TestSuite/suspense/TestEventDetailsCache.ts

```diff
@@ -275,3 +358,189 @@ async function fetchAndCachePossibleDomNode(
+  const isUnderTest = false;
+  // Playwright can be used from multiple languages
+  const sdkLanguage = "javascript";
+  // data-testid can be customized by the user
+  const testIdAttributeName = "data-testid";
+  // WebKit on Windows needs 5, everything else is 1
+  const rafCountForStablePosition = 1;
+  const browserName = "chromium";
+  const customSelectorEngines: string[] = [];
+
+  const preloadExpression = `
+        (() => {
+          const module = {};
+          ${playwrightInjectedScriptModule.source}
+          return new (module.exports.InjectedScript())(
+            globalThis,
+            ${isUnderTest},
+            "${sdkLanguage}",
+            "${testIdAttributeName}",
+            ${rafCountForStablePosition},
+            "${browserName}",
+            [${customSelectorEngines.join(",\n")}]
+          );
+        })();
+      `;
+
+  const pointsToLocatorStrings: Record<string, string> = {};
+  for (const userEvent of filteredEvents) {
+    // The locator string is always the first command arg.
+    const locatorString = userEvent.data.command.arguments[0];
+    const point = userEvent.data.timeStampedPoints.beforeStep!.point;
+    pointsToLocatorStrings[point] = locatorString;
+  }
+
+  // Any focus range errors here will bubble up to the parent focus cache,
+  // which will retry the load if needed later.
+  await replayClient.runEvaluation(
+    {
+      selector: {
+        kind: "points",
+        points: Object.keys(pointsToLocatorStrings).sort(),
+      },
+      preloadExpressions: [
+        {
+          name: "PLAYWRIGHT_INJECTED_SCRIPT",
+          expression: preloadExpression,
+        },
+      ],
+      expression: `
+        // Inject the lookup table of all locator strings
+        const pointsToLocatorStrings = ${JSON.stringify(pointsToLocatorStrings)};
+
+        const getPlaywrightTestStepDomNodes = ${getPlaywrightTestStepDomNodesString};
+
+        const result = getPlaywrightTestStepDomNodes(pointsToLocatorStrings);
+
+        result;
+    `,
+      fullPropertyPreview: true,
+      limits: { begin, end },
+    },
+    results => {
+      // This logic copy-pasted from AnalysisCache.ts
+      for (const result of results) {
+        // Immediately cache pause ID and data so we have it available for reuse
+        setPointAndTimeForPauseId(result.pauseId, result.point);
+        cachePauseData(replayClient, sources, result.pauseId, result.data);
+      }
+      // Collect them all so we process them in a single batch
+      evalResults.push(...results);
+    }
+  );
+
+  evalResults.sort((a, b) => compareExecutionPoints(a.point.point, b.point.point));
+
+  // We need to reformat the raw analysis data to extract details on the "step details" object.
+  const processedResults: TestEventDetailsEntry[] = await Promise.all(
+    evalResults.map(async result => {
+      const { pauseId, point: timeStampedPoint, returned, data } = result;
```

Comment:
Is this an :any type? Can we use the return type information from `playwrightStepDomNodes.ts` (since it's actually a TS file)?

Reply:
No, `result` is a protocol `RunEvaluationResult`, and `data` is just a standard `Value`.  (In this case, `data` points to a mixed-contents array, but at the type level all we know is it's a `Value`.)